oeps/oep-0045/decisions/0001-tutor-as-replacement-for-edx-configuration.rst

```diff
@@ -0,0 +1,328 @@
+
+Tutor's codebase is technically up-to-date, well organized, and maintained. There is no detectable technical debt.
+
+
+Conclusion of the evaluation
+............................
+
+
+Positives
+`````````
+
+At the time of writing, Tutor is the only existing alternative to ``edx/configuration``. Luckily for the community, it
+ticks a lot of the boxes that qualify a good replacement:
+
+-  It can do single-server production deployments well
+-  It allows for automated deployment as well as, or better than, ``edx/configuration``
+-  It supports Koa, and new releases are added quickly
+-  It allows for theme customization, running edx-platform forks, and installing custom requirements
+-  It is extensible via plugins
+-  Actively maintained by a committed developer; codebase has no technical debt
+-  Licensed under the AGPL v3
+
+
+Omissions
+`````````
+
+There is work to be done before it can be considered a full replacement:
+
+-  There must be a documented way to run highly-available, multi-server production deployments, including a way to run
+   automated rolling upgrades with no downtime
+-  Migration from the native installation must be documented and/or automated
+-  It must support deployments of the master branches of edx-platform and IDAs, and once that comes about it must
+   undergo CI/CD so that breakages can be dealt with quickly
+-  There should be more than one maintainer
+
+It is estimated that initial versions of the above will take anywhere from three to six months to achieve, if sufficient
+developers commit resources to the endeavor. (Part of the test of whether a community-supported deployment method will
+work hinges precisely on whether this community can pool resources to maintain it!)
+
+
+Wishlist
+````````
+
+The following would be great to have once the TODO list above is taken care of, as they're not currently present in
+either Tutor or ``edx/configuration``:
+
+-  Support the deployment of MFEs
+-  There should be a way to reuse the same base LMS image with different themes without having to bake them into the
+   image
+
+
+Decisions
+---------
+
+Adopt Tutor as the community-maintained replacement for ``edx/configuration`` for production deployments, with the
+knowledge that it needs to be `further developed <#omissions>`__ in order to achieve all the described `criteria
+<#criteria>`__.
+
+The above decision will be made public in order to garner support for code contributions.
+
+
+Alternatives
+------------
+
+This document's author could not find another published, open source Open edX deployment project that aims to replace
+``edx/configuration``. While edX is itself moving to `Kubernetes with Helm
+<https://openedx.atlassian.net/wiki/spaces/AC/pages/2107441855/Braindump+on+Configuration+Today+and+Future#Future-with-Containers>`__,
+edX at this time do not intend to make their Helm charts public - though they're open to the idea of publishing and
+maintaining sample versions.
```

Comment:
The following projects also aim at deploying Docker-based Open edX:
https://github.com/openfun/openedx-docker (cc @sampaccoud)
https://github.com/Abstract-Tech/derex.runner/ (cc @silviot)

Reply:
@regisb my 2 cents would be to have a shared official docker image built in a separate repository. Tutor's and openfun/openedx-docker images do very similar installations in the Dockerfile...
The big value behind Tutor is to wrap the whole project in a running environment with a one-click install. This is very valuable for most projects but not for us or other big platforms. 
Some people will deploy in Kubernetes, others in OpenShift, or simply with docker-compose... so it makes sense to have several approaches... but they should all be based on the same official docker images.

Reply:
@regisb: thanks, I was not aware of those.  I'll add them in.

@sampaccoud: while I agree with the sentiment, it is not yet clear how or where these images should be maintained.  Ideally, they would be provided by edX themselves.  They're open to making the Dockerfiles they include in the IDA repositories more flexible.  We should definitely campaign that they do exactly that, while attending to everybody's needs.

Nevertheless, we need to start somewhere.  Particularly, to give them an example of what the community needs.  I'm going to take a look at ``openedx-docker`` as well.